### ISSUE TITLE
Support For Mova M1 dreame.vacuum.r2380

### ISSUE THREAD

**Issue by Tasshack** [Tasshack/dreame-vacuum#607]
**Describe the bug**
Model is in supported devices list but not in the config flow therefore cannot be added to the HA.

https://github.com/Tasshack/dreame-vacuum/discussions/604

**Comment by KingDispo** [comment]
As you recommended: removing the "#" from model number in https://github.com/Tasshack/dreame-vacuum/blob/dev/custom_components/dreame_vacuum/config_flow.py#L123 and restarting HA worked for me. Could now add the Mova M1.

![image](https://github.com/Tasshack/dreame-vacuum/assets/67555214/d4fbf6de-04fb-42ef-b381-8267cc70dd7f)

But then next problem occours:

![image](https://github.com/Tasshack/dreame-vacuum/assets/67555214/83a7b72e-0cfe-4d52-a29d-8935fbfba8ee)

Maybe thats why you remove MOVA M1 - dreame.vacuum.r2380?

**Comment by KingDispo** [comment]
uploading the Full HA log ...

![image](https://github.com/Tasshack/dreame-vacuum/assets/67555214/8a547995-7a9d-4f7f-ab6c-f3f94cf74d4e)
[home-assistant_2024-06-22T23-34-27.619Z.log](https://github.com/user-attachments/files/15942028/home-assistant_2024-06-22T23-34-27.619Z.log)

**Comment by KingDispo** [comment]
Did that.
[home-assistant_2024-06-22T23-34-27.619Z (1).log](https://github.com/user-attachments/files/15942042/home-assistant_2024-06-22T23-34-27.619Z.1.log)

**Comment by KingDispo** [comment]
Here is the new log.

[home-assistant_2024-06-23T00-03-45.580Z.log](https://github.com/user-attachments/files/15942063/home-assistant_2024-06-23T00-03-45.580Z.log)

**Comment by KingDispo** [comment]
Even after overriding the two files the problem still exists:

![image](https://github.com/Tasshack/dreame-vacuum/assets/67555214/f65861f7-6c61-44b9-a61f-77385964fb6c)
[home-assistant_2024-06-23T08-36-18.663Z.log](https://github.com/user-attachments/files/15943871/home-assistant_2024-06-23T08-36-18.663Z.log)

**Comment by KingDispo** [comment]
looks much better now!

![image](https://github.com/Tasshack/dreame-vacuum/assets/67555214/2a2d3701-7e13-4087-8882-d137a5be9de4)

**Comment by KingDispo** [comment]
Map renders correctly, but it changes every seconds to a non good rendered map.
![image](https://github.com/Tasshack/dreame-vacuum/assets/67555214/78741798-ecbb-4bb3-8055-55683ce22146)

Here is a video:

https://github.com/Tasshack/dreame-vacuum/assets/67555214/b563bd74-e4c0-4a0b-a9d9-633128faa8e3

I can not controll the robot via HA.

![image](https://github.com/Tasshack/dreame-vacuum/assets/67555214/b61846b3-d79e-4cb5-adfc-276adca468e0)
If i click on start or click on go to home nothing happens.

**Comment by KingDispo** [comment]
Yes, i can share the robot if you want.

**Comment by KingDispo** [comment]
The Robot is now shared

Am 23.06.2024 um 13:04 schrieb Tasshack ***@***.***>:

﻿

Yes, i can share the robot if you want.

My userid is BP428398. I will fix all the issues and let you know when it is ready to use.

**Comment by KingDispo** [comment]
After some tests i can confirm: controlling the robot via home assistant seems to work. I'm happy after some first tests.

**Comment by KingDispo** [comment]
the Robot is removed from home assistant

**Comment by KingDispo** [comment]
I can try to reset the device. In the instruction papers there is only the reset of the network configuration descriped. Hope that is enough to fix the problem.

Am 24.06.2024 um 09:26 schrieb Tasshack ***@***.***>:

﻿

@KingDispo<https://github.com/KingDispo> I see that your device never goes to sleep mode and also it is acting like it is moving all the time by sending a new map data every second indicates that there is something wrong with it.
Normally integration gets an initial map data from the device and device sends partial map data after that when the map data changes which contains only the changed parts of the map. That means every partial map data must be parsed within their frame order but the integration/app may have some missing frames before the latest partial map. When that happens integration/app requests that specific partial frame from device to fill the gap and parse rest of the partial frames.
Your device skips a frame after the initial map data when sending a partial map and it does not respond to anything when integration requests the missing frame from it (thats why map gets blurred because device will become unavailable) so integration cannot parse the consecutive frames and there is nothing I can do to fix that.

Is it possible for you to hard reset the device and install it again to check the issue has been resolved, otherwise I am not sure what to do here.

**Comment by KingDispo** [comment]
Robot was reset and did his first drive to create first map. After adding to Home Assistant the current map Sensor is refreshed every minute.
Do you think it is possible to deactivate the map features as a workaround? Only controlling the robot via Home Assistant is an important basic feature for me.

**Comment by KingDispo** [comment]
Device is shared with your account again. Thanks for your work and your help.

**Comment by KingDispo** [comment]
Because of performance issues in HA i deleted the robot in home assistant - now everythink is fine again. Unfortunally without the Mova M1 (in HA)

**Comment by KingDispo** [comment]
Thanks for your help and your integration. I sent the device back to the seller. Looking for another vacuum with with fewer problems

**Comment by pezlin** [comment]
I have just purchased a Dreame Mova M1. Can I support in any way?

**Comment by pezlin** [comment]
> > I have just purchased a Dreame Mova M1. Can I support in any way?
> 
> Have you read the whole converstation?

Yes, and to my understanding the issue with the map is still unresolved due to that you had no device to test with. I haven't installed the integration yet but I noticed in the instructions that it was possible to choose "Without map".

**Comment by pezlin** [comment]
Seems to be working now. At least when doing a quick test.

**Comment by crembz** [comment]
In thinking of getting one of these as a secondary vac. Is the integration working?

**Comment by p4ch14** [comment]
It seems that Dreame has made some improvements. Previously (a few months ago), I received a message that there was no compatible device linked to my Dreame account. After removing the "#," the device started to become visible (sometimes it was, sometimes it wasn't), but I couldn't complete the initialization process. Replacing the files with those attached in the thread didn't help, so I gave up on integration and reverted to the original files of the latest beta version 2.0 of Dreame-vacuum.

Somewhere in November, my Mova M1 received a firmware update (currently 1.1.0_1401). So, I tried again, and surprisingly, all I had to do was remove the appropriate "#" in config_flow.py, and not only was the robot found, but it was also successfully initialized and configured without any errors (maps are available, sensor information too). At the moment, it looks like the integration with my Mova M1 (through the Dreame account) is working without major issues.
